aten/src/ATen/native/Convolution.cpp

```diff
@@ -203,35 +217,46 @@ static void check_shape_forward(const at::Tensor& input,
 
   AT_CHECK(weight_dim == k,
            "Expected ", weight_dim, "-dimensional input for ", weight_dim,
-           "-dimensional weight ", weight.sizes(), ", but got ", k, "-dimensional input of size ",
+           "-dimensional weight ", weight_sizes, ", but got ", k, "-dimensional input of size ",
            input.sizes(), " instead");
-  AT_CHECK(weight.size(0) >= groups,
+  AT_CHECK(weight_sizes[0] >= groups,
            "Given groups=", groups, ", expected weight to be at least ", groups,
-           " at dimension 0, but got weight of size ", weight.sizes(), " instead");
-  AT_CHECK(weight.size(0) % groups == 0,
+           " at dimension 0, but got weight of size ", weight_sizes, " instead");
+  AT_CHECK(weight_sizes[0] % groups == 0,
            "Given groups=", groups, ", expected weight to be divisible by ",
-           groups, " at dimension 0, but got weight of size ", weight.sizes(),
+           groups, " at dimension 0, but got weight of size ", weight_sizes,
            " instead");
 
+  AT_CHECK(
+      !(input_is_mkldnn && k != 4),
```

Comment:
is it feasible to move the "non-standard" checks into the mkldnn part?  Obviously you can't do that with contiguous (we can figure out a better solution for that once we have proper dimension order support), but I don't see why things like this need to be here?